custom-elements/throw-on-dynamic-markup-insertion-counter-construct.html

```diff
@@ -0,0 +1,117 @@
+<!DOCTYPE html>
+ <html>
+ <head>
+ <title>Custom Elements: create an element for a token must increment and decrement document's throw-on-dynamic-markup-insertion counter</title>
+ <meta name="author" title="Ryosuke Niwa" href="mailto:rniwa@webkit.org">
+ <meta name="assert" content="Invoking document.open, document.write, document.writeln, and document.write must throw an exception when the HTML parser is creating a custom element for a token">
+ <meta name="help" content="https://html.spec.whatwg.org/multipage/parsing.html#create-an-element-for-the-token">
+ <meta name="help" content="https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#throw-on-dynamic-markup-insertion-counter">
+ <script src="/resources/testharness.js"></script>
+ <script src="/resources/testharnessreport.js"></script>
+ <script src="/resources/custom-elements-helpers.js"></script>
+ </head>
+ <body>
+ <div id="log"></div>
+ <script>
+ 
+ async function construct_custom_element_in_parser(test, call_function)
+ {
+     const window = await create_window_in_test(test);
+     const document = window.document;
+ 
+     document.open();
+ 
+     let executed = false;
+     let exception = null;
+     class CustomElement extends window.HTMLElement {
+         constructor() {
+             super();
+             try {
+                 call_function(document, window);
+             } catch (error) {
+                 exception = error;
+             }
+             executed = true;
+         }
+     }
+     window.customElements.define('some-element', CustomElement);
+ 
+     document.write('<!DOCTYPE html><html><body><some-element></some-element></body></html>');
+     document.close();
+ 
+     assert_true(executed, 'Must synchronously instantiate a custom element');
+     return {window, document, exception};
+ }
+ 
+ promise_test(async function () {
+     const result = await construct_custom_element_in_parser(this, (document) => document.open());
+     assert_throws({name: 'InvalidStateError'}, () => { throw result.exception; }, 'Must throw an InvalidStateError');
```

Comment:
Please use the argument `'InvalidStateError'` rather than `{name: 'InvalidStateError'}` for DOMExceptions. Also elsewhere in this PR.

Reply:
done!